docs/scalar_field.rst

```diff
@@ -0,0 +1,174 @@
+.. _scalar_field:
```

Comment:
Please revert the filename, that make my review hard, I can't view in diff mode if you insist to do so.